tools/state-viewer/src/state_changes.rs

```diff
@@ -0,0 +1,219 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_chain::types::RuntimeAdapter;
+use near_chain::{ChainStore, ChainStoreAccess};
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::hash::CryptoHash;
+use near_primitives::trie_key::TrieKey;
+use near_primitives::types::{
+    EpochId, ShardId, StateChangesForBlock, StateChangesForBlockRange, StateChangesForShard,
+    StateRoot,
+};
+use near_primitives_core::types::BlockHeight;
+use near_store::{KeyForStateChanges, Store, WrappedTrieChanges};
+use nearcore::{NearConfig, NightshadeRuntime};
+use std::path::{Path, PathBuf};
+
+#[derive(clap::Subcommand, Debug, Clone)]
+pub(crate) enum StateChangesSubCommand {
+    /// Applies StateChanges from a file.
+    /// Needs state roots because chunks state roots may not be known.
+    /// Needs BlockHeaders available for the corresponding blocks.
+    Apply {
+        /// Location of the input file.
+        /// The file must be generated by the Dump subcommand.
+        #[clap(parse(from_os_str))]
+        file: PathBuf,
+        /// Id of the shard to apply state changes to.
+        shard_id: ShardId,
+        /// State root of the shard at the height of the first block with state changes.
+        state_root: StateRoot,
+    },
+    /// Dumps state changes for a range of blocks from --height-from to --height-to inclusive.
```

Comment:
so this dumps it for all the shards, right?

Reply:
Right, clarified